src/main/kotlin/Mempool.kt

```diff
@@ -1,19 +1,46 @@
-import kotlinx.serialization.Serializable
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.serialization.kotlinx.json.*
 import kotlinx.serialization.json.Json
-import java.net.URL
-
-
-class MempoolSpace {
-    private val json = Json { ignoreUnknownKeys = true }
 
+suspend fun fetchFirstBlockId(startHeight: Int): String {
```

Comment:
You should:

- use a class to avoid duplicating the common parts (this will also let us introduce an `interface` later to make testing easier)
- don't use `try/catch`, kotlin has better ways of dealing with errors, such as the `Result<T>` type

This would look something like this:

```kotlin
class MempoolClient {
    private val httpClient = HttpClient(CIO) {
        install(ContentNegotiation) {
            json(Json {
                ignoreUnknownKeys = true
                prettyPrint = true
                isLenient = true
            })
        }
    }

    suspend fun fetchFirstBlockId(startHeight: Int): Result<String> {
        val response = httpClient.get("https://mempool.space/api/v1/blocks/$startHeight")
        return response.runCatching {
            body<List<Block>>().first().id
        }
    }

    suspend fun fetchTransactionIds(blockId: String): Result<List<String>> {
        val response = httpClient.get("https://mempool.space/api/block/$blockId/txids")
        return response.runCatching {
            body<List<String>>()
        }
    }
}
```